extensions/wikia/Recirculation/js/helpers/LiftigniterHelper.js

```diff
@@ -5,95 +5,93 @@ define('ext.wikia.recirculation.helpers.liftigniter', [
 ], function ($, w, thumbnailer) {
 	'use strict';
 
-	var helper = function (config) {
-		var defaults = {
-				max: 5,
-				width: 320,
-				height: 180,
-				flush: false
-			},
-			options = $.extend({}, defaults, config);
-
-		function loadData() {
-			var deferred = $.Deferred(),
-				registerOptions = {
-					max: options.max * 2, // We want to load twice as many because we filter based on thumbnails
-					widget: options.widget,
-					callback: function (response) {
-						deferred.resolve(formatData(response));
+	var willFetch = {},
+		fetchQueue = [],
+		fetching = false;
+
+	var prepare = function (options) {
+
+		var deferred = $.Deferred(),
+			registerOptions = {
+				max: options.max,
+				widget: options.widget,
+				callback: function (response) {
+					resizeThumbnails(response, options);
+					deferred.resolve(response);
+					fetching = false;
+					if(fetchQueue.length) {
+						// setTimeout is necessary because LI need to wait for LI to remove already registered modules
+						// LI clear register modules just after running all callbacks
+						setTimeout(function () {
+							fetch(fetchQueue.shift());
+						});
 					}
-				};
-
-			if (!w.$p) {
-				return deferred.reject('Liftigniter library not found').promise();
-			}
+				}
+			};
 
-			if (options.opts) {
-				registerOptions.opts = options.opts;
-			}
+		if (!w.$p) {
+			return deferred.reject('Liftigniter library not found').promise();
+		}
 
-			// Callback renders and injects results into the placeholder.
-			w.$p('register', registerOptions);
+		if (options.opts) {
+			registerOptions.opts = options.opts;
+		}
 
-			if (options.flush) {
-				w.$p('fetch');
-			}
+		var module = {
+			promise: deferred,
+			options: registerOptions
+		};
 
-			return deferred.promise();
+		if(willFetch[options.modelName]) {
+			willFetch[options.modelName].push(module)
+		} else {
+			willFetch[options.modelName] = [module];
 		}
 
-		function formatData(data) {
-			var items = [],
-				title = '';
+		return deferred.promise();
+		// TODO setup tracking
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)